tests/pytests/test_debug_commands.py

```diff
@@ -362,29 +366,285 @@ def testSpecIndexesInfo(env: Env):
+            # In cluster mode, RPNet_next (the upstream rp) checks for timeouts at the end/start of reading each shard's reply.
+            # i.e we will always consume the entire chunk of results from each shard before checking for timeout.
+            # If some shards returns fewer than timeout_res_count results (due to EOF), we will continue to read replies until
+            # the sum of their results exceeds timeout_res_count.
+            # if the sum of results from all shards exceeds timeout_res_count but *each* shard individually returns fewer:
+            # no timeout warning is issued, the cursor is depleted.
+            # if the sum of results from all shards exceeds timeout_res_count but *one* shard returns timeout_res_count:
+            # we will get a timeout warning, and a valid cursor id.
+            remaining = limit - total_returned
+            if remaining <= timeout_res_count * env.shardsCount:
+                # we don't know how many results are left in each shard, check only timeout
+                if env.isCluster():
+                    check_res_count = False
+                else:
+                    # in a single shard the next read will return EOF
+                    expected_results_per_iter = remaining
+            res, cursor = env.cmd('FT.CURSOR', 'READ', 'idx', cursor)
+            total_returned += len(res['results'])
+            if cursor == 0:
+                should_timeout = False
+
+            if check_res_count == False:
+                self.verifyWarning(res, f"AggregateDebug with cursor: iter: {iter}, total_returned: {total_returned}", should_timeout=should_timeout)
+            else:
+                self.verifyResultsResp3(res, expected_results_per_iter, f"AggregateDebug with cursor: iter: {iter}, total_returned: {total_returned}", should_timeout=should_timeout)
+            iter += 1
+        env.assertEqual(total_returned, self.num_docs, message=f"AggregateDebug with cursor: depletion took {iter} iterations")
+
+        # cursor count smaller than timeout count, expect no timeout
+        cursor_count = timeout_res_count // 2
+        cursor_query = [*basic_debug_query, 'WITHCURSOR', 'COUNT', cursor_count]
+        res, cursor = env.cmd(*cursor_query, 'LIMIT', 0, limit, *debug_params)
+        should_timeout = False
+        self.verifyResultsResp3(res, cursor_count, should_timeout=should_timeout, message="AggregateDebug with cursor count lower than timeout_res_count:")
+
+        self.StrictPolicy()
+
+    def testAggregateDebug(self):
+        self.AggregateDebug()
+
+    def testAggregateDebug_MT(self):
+        self.env.expect(config_cmd(), 'SET', 'WORKERS', 4).ok()
+        self.AggregateDebug()
+        self.env.expect(config_cmd(), 'SET', 'WORKERS', 0).ok()
+
+    # compare results of regular query and debug query
+    def Sanity(self, cmd, query_params):
+        # avoid running this test in cluster mode, as it relies on the order of the results
+        skipTest(cluster=True)
```

Comment:
I think this sanity is important in cluster mode as well, can we test the results *set* is as expected in this case?

Reply:
Sorry, bad choice of words - i meant order of replies from the shards
Also, we actually "cheat" here in some sens, as a sorter query with timeout potentially returns different results compared to a query that is able to iterate the entire index (no timeout).
(in this case it works because the order of docs insertion was sorted so the docs we get are already sorted)
I prefer not to complicate this test. We have enough coverage for different scenarios in cluster mode.